core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -53,19 +52,19 @@ private[spark] class MapOutputTrackerMasterActor(tracker: MapOutputTrackerMaster
         val msg = s"Map output statuses were $serializedSize bytes which " +
           s"exceeds spark.akka.frameSize ($maxAkkaFrameSize bytes)."
 
-        /* For SPARK-1244 we'll opt for just logging an error and then throwing an exception.
-         * Note that on exception the actor will just restart. A bigger refactoring (SPARK-1239)
-         * will ultimately remove this entire code path. */
+        /* For SPARK-1244 we'll opt for just logging an error and then sending it to the sender.
+         * A bigger refactoring (SPARK-1239) will ultimately remove this entire code path. */
         val exception = new SparkException(msg)
         logError(msg, exception)
-        throw exception
+        context.sendFailure(exception)
```

Comment:
> There are exceptions that cannot be serialized, and we should be careful there.

Because `Throwable implements java.io.Serializable`, I assume all exceptions can be serialized. Do you mean that some exception may throw an exception in `writeObject(java.io.ObjectOutputStream out)` to prevent it from being serialized?